improver_tests/utilities/test_cube_checker.py

```diff
@@ -97,69 +102,70 @@ class Test_check_cube_coordinates(IrisTest):
-        result = check_cube_coordinates(cube, new_cube)
-        self.assertEqual(result.dim_coords, cube.dim_coords)
+        new_cube = self.cube.copy()
+        new_cube.transpose([2, 1, 0])
+        result = check_cube_coordinates(self.cube, new_cube)
+        self.assertEqual(result.dim_coords, self.cube.dim_coords)
 
     def test_coord_promotion(self):
         """Test that scalar coordinates in new_cube are promoted to dimension
         coordinates to match the parent cube."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
-        result = check_cube_coordinates(cube, new_cube)
-        self.assertEqual(result.dim_coords, cube.dim_coords)
+        new_cube = iris.util.squeeze(self.cube)
+        result = check_cube_coordinates(self.cube, new_cube)
+        self.assertEqual(result.dim_coords, self.cube.dim_coords)
 
     def test_coord_promotion_only_dim_coords_in_parent(self):
         """Test that only dimension coordinates in the parent cube are matched
         when promoting the scalar coordinates in new_cube. Here realization is
         made into a scalar coordinate on the parent, and so should remain a
         scalar in new_cube as well."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
-        cube = cube[0]
+        new_cube = iris.util.squeeze(self.cube)
+        cube = self.cube[0]
         result = check_cube_coordinates(cube, new_cube)
         self.assertEqual(result.dim_coords, cube.dim_coords)
 
     def test_coord_promotion_and_reordering(self):
         """Test case in which a scalar coordinate are promoted but the order
         must be corrected to match the progenitor cube."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
-        cube.transpose(new_order=[1, 0, 2, 3])
-        result = check_cube_coordinates(cube, new_cube)
-        self.assertEqual(result.dim_coords, cube.dim_coords)
+        new_cube = iris.util.squeeze(self.cube)
+        self.cube.transpose(new_order=[1, 0, 2])
+        result = check_cube_coordinates(self.cube, new_cube)
+        self.assertEqual(result.dim_coords, self.cube.dim_coords)
 
     def test_permitted_exception_coordinates(self):
         """Test that if the new_cube is known to have additional coordinates
         compared with the original cube, these coordinates are listed are
         exception_coordinates and handled correctly."""
-        cube = set_up_cube()
-        new_cube = cube[0].copy()
-        cube = iris.util.squeeze(cube)
-        exception_coordinates = ["time"]
+        new_cube = self.cube.copy()
+        cube = iris.util.squeeze(self.cube)
+        exception_coordinates = ["realization"]
         result = check_cube_coordinates(
             cube, new_cube, exception_coordinates=exception_coordinates
         )
-        dim_coords = tuple(new_cube.coord("time")) + cube.dim_coords
+        dim_coords = tuple(new_cube.coord("realization")) + cube.dim_coords
```

Comment:
If you keep this using `exception_coordinates = ["time"]`, you save all the new setup and cube squeezing.

Reply:
One could promote time to be a dimension coordinate on the reference cube, but as it already has realization I have stuck with that. I have removed all the calls to squeeze by creating a top level squeezed cube which simplifies things as I think you wanted.